public/jsx/views/card-details.jsx

```diff
@@ -0,0 +1,56 @@
+'use strict';
+
+var $ = require('jquery');
+var React = require('react');
+
+var CardForm = require('components/card-form');
+var Spinner = require('components/spinner');
+
+
+module.exports = React.createClass({
+
+  displayName: 'CardDetailsView',
+
+  getInitialState: function() {
+    return {
+      braintree_token: false,
+    };
+  },
+
+  propTypes: {
+    apiSource: React.PropTypes.string.isRequired,
+  },
+
+  getDefaultProps: function() {
+    return {
+      apiSource: '/api/braintree/token/generate/',
+    };
+  },
+
+  componentDidMount: function() {
+    $.ajax({
+      method: 'post',
+      url: this.props.apiSource,
+      context: this,
+    }).then(function(data) {
+      if (this.isMounted()) {
+        this.setState({'braintree_token': data.token}); // eslint-disable-line
+      }
```

Comment:
It's more that the best practice is not to change the state of something if it's unmounted. That would mean the component was destroyed.